numpy/core/tests/test_multiarray.py

```diff
@@ -4728,6 +4728,16 @@ def test_vecself(self):
         p = np.inner(a, a)
         assert_almost_equal(p, 0, decimal=14)
 
+    def test_inner_product_with_various_contiguities(self):
+        # github issue 6532
+        for dt in int, float:
+            A = np.array([[1, 2], [3, 4]], dtype=dt)
+            B = np.array([[1, 3], [2, 4]], dtype=dt)
+            C = np.array([1, 1], dtype=dt)
+            assert_equal(A.T, B)
```

Comment:
By the way, I'm pretty sure I've also seen the opposite suggestion in numpy reviews regarding assert vs. testing.assert_*